source/huffman.c

```diff
@@ -114,6 +114,30 @@ static int encode_write_bit_pattern(struct encoder_state *state, struct aws_huff
     return AWS_OP_SUCCESS;
 }
 
+size_t aws_huffman_get_encoded_length(struct aws_huffman_encoder *encoder, struct aws_byte_cursor to_encode) {
+
+    AWS_PRECONDITION(encoder);
+    AWS_PRECONDITION(to_encode.ptr && to_encode.len);
+
+    size_t num_bits = 0;
+
+    while (to_encode.len) {
+        uint8_t new_byte = 0;
+        aws_byte_cursor_read_u8(&to_encode, &new_byte);
+        struct aws_huffman_code code_point = encoder->coder->encode(new_byte, encoder->coder->userdata);
+        num_bits += code_point.num_bits;
+    }
+
+    size_t length = num_bits / 8;
```

Comment:
just a thought:

how hot is this path? if it is called when we need to be fast, you can change these to bit operations:

num_bits >> 3

and 

if (num_bits & 0x07)

Reply:
Check the disasm, I will bet you anything that compilers are already doing this.

Reply:
BOY DO I HAVE NEWS FOR YOU https://godbolt.org/z/3df92J

tl;dr: the compiler does this for me, and I think the way it is now is easier to read.